plugins/hydra_nevergrad_sweeper/tests/test_nevergrad_sweeper_plugin.py

```diff
@@ -0,0 +1,63 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+# noinspection PyUnresolvedReferences
+from typing import Any
+
+import nevergrad as ng  # type: ignore
+import pytest
+
+from hydra.core.plugins import Plugins
+from hydra.plugins.sweeper import Sweeper
+from hydra.test_utils.test_utils import TSweepRunner, sweep_runner  # noqa: F401
+from hydra_plugins.hydra_nevergrad_sweeper import NevergradSweeper
+from hydra_plugins.hydra_nevergrad_sweeper.core import make_parameter
+
+
+def test_discovery() -> None:
+    # Tests that this plugin can be discovered via the plugins subsystem when looking at the Sweeper plugins
+    assert NevergradSweeper.__name__ in [x.__name__ for x in Plugins.discover(Sweeper)]
+
+
+@pytest.mark.parametrize(  # type: ignore
+    "string,param_cls,value_cls",
+    [
+        ("blu,blublu", ng.p.Choice, str),
+        ("0,1,2", ng.p.TransitionChoice, int),
+        ("0.0,12.0,2.0", ng.p.Choice, float),
+        ("1:12", ng.p.Scalar, int),
+        ("1.0:12", ng.p.Scalar, float),
+        ("blublu", str, str),
+        (
+            "Scalar(init=12.1).set_mutation(sigma=3).set_integer_casting()",
+            ng.p.Scalar,
+            int,
+        ),
+    ],
+)
+def test_make_parameter(string: str, param_cls: Any, value_cls: Any) -> None:
+    param = make_parameter(string)
+    assert isinstance(param, param_cls)
+    if param_cls is not str:
+        assert isinstance(param.value, value_cls)
+
+
+# pylint: disable=redefined-outer-name
+def test_launched_jobs(sweep_runner: TSweepRunner) -> None:  # noqa: F811 # type: ignore
```

Comment:
well done.

Can you also add a test that specifically tests the example app?

`chdir_plugin_root()` on master (newly added) can be used to ensure that your working directory is set to the root of the plugin in the test. (see it's usage in the joblib plugin test).

Reply:
I updated the test, but did not succeed in making it work correctly: it does not seem to actually call the "dummy_training" function, so it does not optimize anything. Any insights on why? (I've added `chdir_plugin_root()` with no improvement)

Reply:
By the way, not sure if you rebased on top or not but there are recent changes to how nox is doing plugin coverage which might interact with this.

I will take a look at the test output to see if I can figure out the cause.

Reply:
There are a few ways to test sweeps right now.
using the sweep_runner or calling the executable directly.

calling_module and calling_file in the sweep runner are used to determine what the config path is relative to.
you want to override the task_function if you want to have custom logic here. I think you can just make task_function the dummy_training function.

The second way is to call the sweep script directly, this is more of an integration end to end test.
For examples I generally try to do that to ensure that the actual example is not breaking (I think the example I gave you initially was for the second style though).

You can see some similar examples in several tests.
This is how we test the output of the ax plugin:
```
def test_ax_logging(tmpdir: Path) -> None:
    cmd = [
        sys.executable,
        "tests/apps/polynomial.py",
        "-m",
        "hydra.run.dir=" + str(tmpdir),
        "polynomial.x=-5:-2",
        "polynomial.y=-1,1",
        "polynomial.z=10",
        "hydra.sweeper.params.ax_config.max_trials=3",
    ]
    result = subprocess.check_output(cmd).decode("utf-8").rstrip()
    assert "polynomial.x: range=[-5, -2], type = int" in result
    assert "polynomial.y: choice=[-1, 1], type = int" in result
    assert "polynomial.z: fixed=10, type = int" in result
```

You can do something similar as well (or instead of using sweep_runner).